### PR TITLE
chore(risedev): make risedev passthrough RUST_LOG env

### PR DESCRIPTION
**Pull request by MrCroxx**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Make `risedev` passthrough `RUST_LOG` env for debugging.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
IIUC, env should always be passthrough as `shell -> cargo-make -> risedev -> tmux -> risingwave`. For example, I test the distributed tracing with `RW_TRACING_ENDPOINT=... ./risedev d` every day. 🤔

**Comment by BugenZhao** [comment]
Also in e2e tests:

https://github.com/risingwavelabs/risingwave/blob/adfab42d9a9b8699c49f79e1397288be4f6dbd42/ci/scripts/run-e2e-test.sh#L149-L150

----

How do you find it's necessary?

**Comment by MrCroxx** [comment]
> IIUC, env should always be passthrough as `shell -> cargo-make -> risedev -> tmux -> risingwave`. For example, I test the distributed tracing with `RW_TRACING_ENDPOINT=... ./risedev d` every day. 🤔

I use `RUST_LOG=xxx ./risedev d` to setup a playground but the `RUST_LOG` is not passed, I checked it in risingwave code and the welcome window, neither get the env. 🤔

**Comment by StrikeW** [comment]
> > IIUC, env should always be passthrough as `shell -> cargo-make -> risedev -> tmux -> risingwave`. For example, I test the distributed tracing with `RW_TRACING_ENDPOINT=... ./risedev d` every day. 🤔
> 
> I use `RUST_LOG=xxx ./risedev d` to setup a playground but the `RUST_LOG` is not passed, I checked it in risingwave code and the welcome window, neither get the env. 🤔

Have you try `export RUST_LOG=xxx ./risedev d`?

**Comment by MrCroxx** [comment]
> Have you try export RUST_LOG=xxx ./risedev d?

Yep. Still not working. But tmux DOES passed other envs, which is wired. I'm investigating.

**Comment by StrikeW** [comment]
> > Have you try export RUST_LOG=xxx ./risedev d?
> 
> Yep. Still not working. But tmux DOES passed other envs, which is wired. I'm investigating.

Maybe related: https://github.com/risingwavelabs/risingwave/issues/12922#issuecomment-1767694633

**Comment by MrCroxx** [comment]
With MacOS tmux 3.3a, the env is correctly passed:

```bash
> uname -ar
Darwin mba 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:59:33 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8112 arm64 arm Darwin
> tmux -V
tmux 3.3a
```

But with Ubuntu tmux 3.2a, it cannot:

```bash
> uname -ar
Linux homelab 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
> tmux -V
tmux 3.2a
```

**Review by wangrunji0408 — APPROVED** [review]

**Review by tabVersion — APPROVED** [review]

**Comment by MrCroxx** [comment]
Ubuntu with tmux 3.3a still not working. 🤔

```
> uname -ar
Linux homelab 6.2.0-39-generic #40-Ubuntu SMP PREEMPT_DYNAMIC Tue Nov 14 14:18:00 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
> lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 23.04
Release:	23.04
Codename:	lunar
> tmux -V
tmux 3.3a
```

**Comment by MrCroxx** [comment]
> Also in e2e tests:
> 
> https://github.com/risingwavelabs/risingwave/blob/adfab42d9a9b8699c49f79e1397288be4f6dbd42/ci/scripts/run-e2e-test.sh#L149-L150
> 
> How do you find it's necessary?

Let me check if env here really works with CI, the default log level is already info.

**Comment by MrCroxx** [comment]
I think CI cannot pass env without this PR:

With [`99f8908` (#14228)](https://github.com/risingwavelabs/risingwave/pull/14228/commits/99f8908108cfab0bd2abe0e587f8db9fab38ca1e) :

```
RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" TTT="1" \
    cargo make ci-start ci-compaction-test
```

```
[tasks.ci-check]
category = "RiseDev - CI"
dependencies = ["clean-data", "pre-start-dev"]
script = """
echo $RUST_LOG
echo $TTT
"""

[tasks.ci-start]
category = "RiseDev - CI"
# ... ...
```

Got:

```
t=1703668391646 info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info

t=1703668391646 
```

**Comment by st1page** [comment]
works on arch with tmux 3.3a
```
~/risingwave on  sts/row_id_gen_handle_invis_delete ⌚ 17:28:14
$ uname -ar
Linux stonepage 6.6.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 14 Dec 2023 03:45:42 +0000 x86_64 GNU/Linux

~/risingwave on  sts/row_id_gen_handle_invis_delete ⌚ 17:28:46
$ tmux -V       
tmux 3.3a

```

**Comment by BugenZhao** [comment]
Is this snippet related?

https://github.com/risingwavelabs/risingwave/blob/adfab42d9a9b8699c49f79e1397288be4f6dbd42/Makefile.toml#L85-L92

**Comment by MrCroxx** [comment]
> Is this snippet related?
> 
> https://github.com/risingwavelabs/risingwave/blob/adfab42d9a9b8699c49f79e1397288be4f6dbd42/Makefile.toml#L85-L92

I think not. I tried other env and still not working. With working setup (e.g. MacOS), it still works.

**Comment by BugenZhao** [comment]
> I tried other env and still not working.

If so, I guess we have to either finding out what's on earth the problem is, or passthrough all environment variables. 😕

File: Makefile.toml [hunk 1]
```diff
@@ -1298,9 +1298,17 @@ echo RiseDev installed to $(tput setaf 4)${INSTALL_PATH}$(tput sgr0)
...
 """
 
-[tasks.ci-start]
+[tasks.ci-check]
 category = "RiseDev - CI"
 dependencies = ["clean-data", "pre-start-dev"]
+script = """
+echo $RUST_LOG
+echo $TTT
+"""
```
Comment (yuhao-su) [inline Makefile.toml:1307]: Can you try specify a shell here?

**Review by Li0k — APPROVED** [review]

**Comment by xxchan** [comment]
I cannot reproduce 😟
image

**Comment by BugenZhao** [comment]
May I ask whether we have identified where the issue is?

**Pull request closed without merging**